packages/world/src/modules/core/CoreModule.sol

```diff
@@ -97,34 +109,62 @@ contract CoreModule is Module {
   }
 
   /**
-   * @notice Register the CoreSystem in the World.
-   * @dev Uses the CoreSystem's `registerSystem` implementation to register itself on the World.
+   * @notice Register the core systems in the World.
+   */
+  function _registerCoreSystems() internal {
+    _registerSystem(accessManagementSystem, ACCESS_MANAGEMENT_SYSTEM_ID);
+    _registerSystem(balanceTransferSystem, BALANCE_TRANSFER_SYSTEM_ID);
+    _registerSystem(batchCallSystem, BATCH_CALL_SYSTEM_ID);
+    _registerSystem(coreRegistrationSystem, CORE_REGISTRATION_SYSTEM_ID);
+  }
+
+  /**
+   * @notice Register the core system in the World.
+   * @dev Uses the CoreRegistrationSystem's `registerSystem` implementation to register the system on the World.
    */
-  function _registerCoreSystem() internal {
+  function _registerSystem(address target, ResourceId systemId) internal {
     WorldContextProviderLib.delegatecallWithContextOrRevert({
       msgSender: _msgSender(),
       msgValue: 0,
-      target: coreSystem,
-      callData: abi.encodeCall(WorldRegistrationSystem.registerSystem, (CORE_SYSTEM_ID, CoreSystem(coreSystem), true))
+      target: coreRegistrationSystem,
+      callData: abi.encodeCall(WorldRegistrationSystem.registerSystem, (systemId, WorldContextConsumer(target), true))
```

Comment:
re: above, this is the bit that confused me and makes me wonder if we should just have all the systems registered individually rather than grouping some under `CoreRegistrationSystem`

it was a little unclear that `CoreRegistrationSystem` is a `WorldRegistrationSystem` and we're doing a delegatecall using the `WorldRegistrationSystem` interface rather than `CoreRegistrationSystem` interface